packages/tailwindcss/src/utilities.ts

```diff
@@ -1162,29 +1148,33 @@ export function createUtilities(theme: Theme) {
    * @css `scale`
    */
   staticUtility('scale-none', [['scale', 'none']])
-  utilities.functional('scale', (candidate) => {
-    if (!candidate.value || candidate.modifier) return
+  function handleScale({ negative }: { negative: boolean }) {
+    return (candidate: Extract<Candidate, { kind: 'functional' }>) => {
+      if (!candidate.value || candidate.modifier) return
 
-    let value
-    if (candidate.value.kind === 'arbitrary') {
-      value = candidate.value.value
-      return [decl('scale', value)]
-    } else {
-      value = theme.resolve(candidate.value.value, ['--scale'])
-      if (!value && isPositiveInteger(candidate.value.value)) {
-        value = `${candidate.value.value}%`
+      let value
+      if (candidate.value.kind === 'arbitrary') {
+        value = candidate.value.value
+        return [decl('scale', value)]
+      } else {
+        value = theme.resolve(candidate.value.value, ['--scale'])
+        if (!value && isPositiveInteger(candidate.value.value)) {
+          value = `${candidate.value.value}%`
+        }
+        if (!value) return
       }
-      if (!value) return
+      value = negative ? `calc(${value} * -1)` : value
+      return [
+        scaleProperties(),
+        decl('--tw-scale-x', value),
+        decl('--tw-scale-y', value),
+        decl('--tw-scale-z', value),
+        decl('scale', `var(--tw-scale-x) var(--tw-scale-y)`),
+      ]
     }
-    value = withNegative(value, candidate)
-    return [
-      scaleProperties(),
-      decl('--tw-scale-x', value),
-      decl('--tw-scale-y', value),
-      decl('--tw-scale-z', value),
-      decl('scale', `var(--tw-scale-x) var(--tw-scale-y)`),
-    ]
-  })
+  }
+  utilities.functional('-scale', handleScale({ negative: true }))
+  utilities.functional('scale', handleScale({ negative: false }))
```

Comment:
Could these use `functionalUtility`?

Reply:
Maybe if we apply some gymnastics. But I think we didn't use `functionalUtility` before because we register `scale-x`, `scale-y` and `scale-z` as separate functional utilities as well. If we make `scale` a functional utility, then candidates such as `scale-x-50` will be handled by the `scale` functional utility with a value of `x-50`.

Reply:
Looked into it quickly, looks like the real reason we don't do it this way is because we handle arbitrary values differently than regular values — `scale-[4_6]` just produces `scale: 4 6` and bypasses all the composition variables and stuff.

Reply:
That is also true. We could handle that when handling the functional utility. 

Don't think that's worth it 🤔